Amplify/Categories/DataStore/Subscribe/DataStoreCategory+Subscribe.swift

```diff
@@ -12,4 +12,12 @@ extension DataStoreCategory: DataStoreSubscribeBehavior {
     public func publisher<M: Model>(for modelType: M.Type) -> AnyPublisher<MutationEvent, DataStoreError> {
         return plugin.publisher(for: modelType)
     }
+
+    @available(iOS 13.0, *)
+    public func observeQuery<M: Model>(for modelType: M.Type,
+                                       where predicate: QueryPredicate? = nil,
+                                       sort sortInput: QuerySortInput? = nil)
```

Comment:
Is there any way that we can have a function that accepts `[QuerySortDescriptor]?` as `sortInput` as well? Today, the `storageEngine.query` function ultimately accepts `[QuerySortDescriptor]?` and so during the Flutter deserialization, the sort is build directly as that - bypassing the `QuerySortInput` struct altogether.